sui_core/src/gateway_state.rs

```diff
@@ -53,6 +56,123 @@ const MAX_TX_RANGE_SIZE: u64 = 4096;
 /// Number of times to retry failed TX
 const MAX_NUM_TX_RETRIES: usize = 5;
 
+/// Prometheus metrics which can be displayed in Grafana, queried and alerted on
+pub struct GatewayMetrics {
+    total_tx_processed: IntCounter,
+    total_tx_errored: IntCounter,
+    num_tx_publish: IntCounter,
+    num_tx_movecall: IntCounter,
+    num_tx_splitcoin: IntCounter,
+    num_tx_mergecoin: IntCounter,
+    total_tx_retries: IntCounter,
+    shared_obj_tx: IntCounter,
+    pub total_tx_certificates: IntCounter,
+    pub num_signatures: Histogram,
+    pub num_good_stake: Histogram,
+    pub num_bad_stake: Histogram,
+    pub transaction_latency: Histogram,
+}
```

Comment:
I actually really like defining these as a struct. its a great way to encapsulate all the metrics a particular component needs.